apps/backoffice-v2/src/domains/collection-flow/hooks/mutations/useUpdateCollectionFlowStateMutation/useUpdateCollectionFlowStateMutation.ts

```diff
@@ -0,0 +1,41 @@
+import { updateCollectionFlowState } from '@/domains/collection-flow/fetchers';
+import { TCollectionFlowState } from '@/domains/collection-flow/schemas';
+import { queryClient } from '@/lib/react-query/query-client';
+import { useMutation } from '@tanstack/react-query';
+import { toast } from 'sonner';
+import { t } from 'i18next';
+import { collectionFlowQueryKeys } from '@/domains/collection-flow/query-keys';
+import { isErrorWithMessage } from '@ballerine/common';
+
+export const useUpdateCollectionFlowStateMutation = () => {
+  return useMutation({
+    mutationFn: ({
+      workflowId,
+      state,
+    }: {
+      workflowId: string;
+      state: TCollectionFlowState;
+      action: 'step_request' | 'step_cancel';
+    }) => {
+      return updateCollectionFlowState(workflowId, state);
+    },
+    onSuccess: (_, { action }) => {
+      toast.success(t(`toast:${action}.success`));
+    },
+    onError: (error, { action }) => {
+      if (isErrorWithMessage(error)) {
+        const translatedError = t(`toast:${action}.error`, {
+          errorMessage: (error as Error).message,
```

Comment:
Using isErrorWithMessage means as Error is no longer needed. If you wish it can be used directly inline on errorMessage too and if it has no message pass some string or whatever you wish. Tl'dr you can safetly remove the type assertion.